include/envoy/network/dns.h

```diff
@@ -24,6 +25,16 @@ class ActiveDnsQuery {
   virtual void cancel() PURE;
 };
 
+struct DnsResponse {
+  DnsResponse(Address::InstanceConstSharedPtr address, const std::chrono::seconds& ttl)
+      : address_(address), ttl_(ttl) {}
+
+  Address::InstanceConstSharedPtr address_;
+  std::chrono::seconds ttl_;
+};
+
+typedef std::shared_ptr<const DnsResponse> DnsResponseSharedPtr;
```

Comment:
convention: DnsResponseConstSharedPtr
BTW: could you double check if a shared pointer of `DnsResponse` is needed? It seems copy by value is more appropriate